### PR TITLE
Fix missing `docs` config files

### LINKED ISSUE #25: [BUG] Missing sphinx configuration
## Description
Sphinx configuration is missing for auto-doc.

Correspondence: @sahilbhola14 
Branch (if applicable):  #18 
### Steps to reproduce issue
[create_documentation.sh](https://github.com/PinguDevTeam/ICF-MI-framework/blob/main/scripts/create_documentation.sh) will given error when generating the documentation

### Current behavior
Missing configuration
…

### Expected behavior
docs folder should be present 
…

### PR DESCRIPTION
**Pull request by sahilbhola14**
## Purpose
-  Doc configs were missing in #18 which would potentially result in failed doc building.
-  This PR also restructures the directory structure to hierarchical.

 Closes #25 

Sphinx config files were missing in #18 CI which would potentially fail the doc building.
Before this PR, this directory structure was modular. This PR introduces hierarchical directory structure.

## Type of change

- Bugfix (non-breaking change which fixes an issue)
- Repository chores (changes made to improve repository/workflow quality)

## Checklist
_Put an `x` in the boxes that apply._

- [x] I have addressed all the issues entirely

### CODE REVIEW

File: docs/introduction.rst [hunk 1]
```diff
@@ -0,0 +1,29 @@
+Introduction
+^^^^^^^^^^^^^
+**ICF-MI** combines two statistical algorithms, **TT-ICE** which is a tensor-train decomposition alogorithm and **GIM** which is a model identification alogorithm.
+The Algorithm has the following salient features:
+
+* **TT-ICE** is a tensor-train decomposition algorithm which is a generalization of the popular **ICE** algorithm. It is a non-parametric algorithm which can be used to decompose a high-dimensional function into a sum of low-dimensional functions.
```
Comment (dorukaks) [inline docs/introduction.rst]: Remind me to correct those :smile:  , ICE stands for [Incremental Core Expansion](https://arxiv.org/abs/2211.12487). Lmk if you want me do the changes before or after merging this PR.

**Comment by dorukaks** [comment]
Let us know which one you are going to merge first #28 or #29

**Comment by sahilbhola14** [comment]
#28 needs to be merged first

File: docs/introduction.rst (hunk 1 above)
Comment (dorukaks) [reply docs/introduction.rst]: If you want those addressed, I'll quickly checkout the branch and do the changes in the morning

**Comment by dorukaks** [comment]
And @sahilbhola14 please don't hesitate to force any directory structure through this PR, we can close PR #26 without merging and reopen to comply with the discussed repository and branching structure.

Hopefully #28 and #29 will be the last PR's into main and we can start working on dev branch while creating our individual features

**Comment by sahilbhola14** [comment]
@dorukaks Okay, I will modify the directory structure as well in this PR

**Review by dorukaks — APPROVED** [review]
As far as I've seen, this looks good. Let's create dev branch after merging this and #28. Let's gooo

File: docs/introduction.rst (hunk 1 above)
Comment (sahilbhola14) [reply docs/introduction.rst]: Oh, Lol. You can make the changes by checking the branch associated with this pull request, i.e. `fix/sahil/25`. Let me know when you make your commit

Comment (dorukaks) [reply docs/introduction.rst]: Updated my part :+1:  good to go

**Review by camckenn — APPROVED** [review]

**Review by cfkawa — APPROVED** [review]

**Comment by dorukaks** [comment]
@cfkawa please refer to the [guideline](https://github.com/PinguDevTeam/ICF-MI-framework/blob/main/formats/branching.md#pull-requests-prs) for further information on how to merge PRs.

**Pull request merged**